pkg/instrumentors/bpf/database/sql/bpf/probe.bpf.c

```diff
@@ -0,0 +1,90 @@
+#include "arguments.h"
+#include "span_context.h"
+#include "go_context.h"
+#include "go_types.h"
+
+char __license[] SEC("license") = "Dual MIT/GPL";
+
+#define MAX_QUERY_SIZE 100
+#define MAX_CONCURRENT 50
+
+struct sql_request_t {
+    u64 start_time;
+    u64 end_time;
+    char query[MAX_QUERY_SIZE];
+    struct span_context sc;
+    struct span_context psc;
+};
+
+struct {
+	__uint(type, BPF_MAP_TYPE_HASH);
+	__type(key, void*);
+	__type(value, struct sql_request_t);
+	__uint(max_entries, MAX_CONCURRENT);
+} context_to_sql_events SEC(".maps");
+
+struct {
+	__uint(type, BPF_MAP_TYPE_PERF_EVENT_ARRAY);
+} events SEC(".maps");
+
+// Injected in init
+volatile const bool should_include_db_statement;
+
+// This instrumentation attaches uprobe to the following function:
+// func (db *DB) queryDC(ctx, txctx context.Context, dc *driverConn, releaseConn func(error), query string, args []any)
+SEC("uprobe/queryDC")
+int uprobe_queryDC(struct pt_regs *ctx) {
+    // argument positions
+    u64 context_ptr_pos = 3;
```

Comment:
I think these should be checked by the offset process and injected like the HTTP param offsets. Offsets can change between version.

Reply:
This context argument is part of the instrumented function signature - which I think we assume is constant. The cases where we use the offset tracker are for struct fields. In the http case the context is part of the Request struct and not passed "directly" as an argument